internal/clients/influxdb.go

```diff
@@ -0,0 +1,54 @@
+/*
+Copyright 2021 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package clients
+
+import (
+	"context"
+
+	"github.com/crossplane/crossplane-runtime/pkg/resource"
+	influxdbv2 "github.com/influxdata/influxdb-client-go/v2"
+	"github.com/pkg/errors"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/crossplane-contrib/provider-influxdb/apis/v1alpha1"
+)
+
+const (
+	errTrackPCUsage = "cannot track ProviderConfig usage"
+	errGetPC        = "cannot get referenced ProviderConfig"
+	errGetCreds     = "cannot get credentials"
+)
+
+// NewClient returns the base InfluxDB client.
+func NewClient(ctx context.Context, kube client.Client, mg resource.Managed) (influxdbv2.Client, error) {
```

Comment:
I would prefer not to pass all these parameters including the client down to the NewClient call here. I feel like, it would be cleaner if we just get the `token` here and Controllers `Connect` handles fetching secret, creating usage tracker etc.

Reply:
I looked at provider-aws and provider-azure to see what we do and it seems like we try to consolidate everything that is common to all managed resources into one function. I think duplication is OK in a lot of cases but here it would scale with the number CRDs so I'm kind of leaning towards keeping it like that.